lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types.find { |type| type.name == type_name }
```

Comment:
Right, that code example doesn't make sure.  Actually, it sounds like you are describing what I said originally: "build and memoize a hash mapping name to type instead".  E.g. have a types_by_name method that is memoized, then you could call `types_by_name.fetch(type_name)` for fast lookups.